app/src/main/java/ee/ioc/phon/android/speechutils/ContinuousRawAudioRecorder.java

```diff
@@ -0,0 +1,215 @@
+package ee.ioc.phon.android.speechutils;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import ee.ioc.phon.android.speechutils.utils.AudioUtils;
+
+public class ContinuousRawAudioRecorder extends AbstractAudioRecorder {
+
+    private static final int DEFAULT_BUFFER_LENGTH_IN_MILLIS = 2000;
+    private static final String LOG_FILTER = "continuous-recorder: ";
+
+    private SessionStartPointer mSessionStartPointer = SessionStartPointer.beginningOfBuffer();
+    private AtomicBoolean mRecordingToFile = new AtomicBoolean(false);
+
+    public static class SessionStartPointer {
+
+        private int mSessionStartPointerMillis;
+        private static SessionStartPointer mBeginningOfBufferPosition = new SessionStartPointer(Integer.MIN_VALUE);
+        private static SessionStartPointer mNowPosition = new SessionStartPointer(0);
+
+        private SessionStartPointer(int sessionStartPointerMillis) {
+            setSessionStartPointerMillis(sessionStartPointerMillis);
+        }
+
+        private void setSessionStartPointerMillis(int sessionStartPointerMillis) {
+            // how many millis we should go back should always be a positive number
+            mSessionStartPointerMillis = Math.abs(sessionStartPointerMillis);
+        }
+
+        int getSessionStartPointerMillis() {
+            // in case that no one set the buffer length, return start pointer as now
+            if (this == mBeginningOfBufferPosition && mSessionStartPointerMillis == Integer.MIN_VALUE)
+                return mNowPosition.getSessionStartPointerMillis();
+
+            // in case that the requested start pointer is bigger than the buffer, return the buffer size
+            if (this != mBeginningOfBufferPosition && mSessionStartPointerMillis > mBeginningOfBufferPosition.getSessionStartPointerMillis())
+                return mBeginningOfBufferPosition.getSessionStartPointerMillis();
+
+            return mSessionStartPointerMillis;
+        }
+
+        static void setRecordingBufferLengthMillis(int recordingBufferLengthMillis) {
+            mBeginningOfBufferPosition.setSessionStartPointerMillis(recordingBufferLengthMillis);
+        }
+
+        public static SessionStartPointer beginningOfBuffer() {
+            return mBeginningOfBufferPosition;
+        }
+
+        public static SessionStartPointer now() {
+            return mNowPosition;
+        }
+
+        public static SessionStartPointer someMillisBack(int millisBackToStartTheSessionFrom) {
+            return new SessionStartPointer(millisBackToStartTheSessionFrom);
+        }
+
+        public static SessionStartPointer someSecondsBack(int secondsBackToStartTheSessionFrom) {
+            return new SessionStartPointer(secondsBackToStartTheSessionFrom * 1000);
+        }
+    }
+
```

Comment:
This is, in most, a copy paste from the code in RawAudioRecorder. I didn't think of a better way to do it with the current encapsulation of the code. It seems to me that there are some gaps in the design of the classes that disables us from inheriting from AbstractAudioRecorder without changing the code there or duplicating code in the child objects. Any idea is welcomed of course in this area. I had a thought about inheriting from RawAudioRecorder instead of AbstractAudioRecorder but this meant adding the continuous flags to the RawAudioRecorder and delegate it from there to the abstract class. Didn't like that. Ideas are welcomed